core/src/main/resources/dev/process_supported_files.py

```diff
@@ -0,0 +1,178 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""
+Description:
+    This Python script goes over all directories of different spark versions under the
+    spark-rapids/tools/generated_files folder and contruct unions of the supported csv
+    files (supportedDataSource.csv, supportedExecs.csv, and supportedExprs.csv).
+
+Dependencies:
+    - numpy >= 1.23.3
+    - pandas >= 2.0.3
+
+Usage:
+    python process_supported_files.py path_to_generated_files_dir
+"""
+
+
+import os
+import argparse
+import pandas as pd
+from enum import IntEnum
+import logging
+
+
+"""
+IntEnum Class for support level types and easy comparison
+"""
+class SupportLevel(IntEnum):
+    NS = 1
+    NA = 2
+    PS = 3
+    S = 4
+    CO = 5
+
+
+"""
+Check if input is in SupportLevel class
+"""
+def is_support_level(name):
+    return name in SupportLevel.__members__
+
+
+"""
+Compare two entries in the CSV file DFs:
+  1) If they are both SupportLevel types, compare using their interger values
+  2) If not, compare their length (assuming strings with greater length contain more info)
+"""
+def is_greater(elem1, elem2):
+    # print("elem1", elem1)
+    # print("elem2", elem2)
+    if is_support_level(elem1) and is_support_level(elem2): 
+        return SupportLevel[elem1] > SupportLevel[elem2]
+    else:
+        return len(elem1) > len(elem2)
```

Comment:
When do we expect to reach this condition?

Reply:
This is condition is used for comparing column which does not have `[S, PS, NA, NS, CO]` values and is not used to define the df row. For instance, column `Notes` in `SupportedExecs.csv` and column `SQL Func` in `supportedExprs.csv`.